File: RepoTransBench/target_projects/Python/C++/zqqqqz2000_shshsh/tests/original/test_pipe_manual.cpp
#include <gtest/gtest.h>
#include <string>
#include <stdexcept>

class Pipe {
public:
    int in_fd = 10, out_fd = 11;
    bool auto_close = true;

    Pipe() {}
    void close_in() { in_fd = -1; }
    void close_out() { out_fd = -1; }
};

class PipeCloseError : public std::runtime_error {
public:
    PipeCloseError() : std::runtime_error("PipeCloseError") {}
    std::string repr() const { return std::string("PipeCloseError()"); }
};

TEST(TestPipeManual, PipeInitAndClose) {
    Pipe pipe;
    ASSERT_TRUE((pipe.in_fd != -1 && pipe.out_fd != -1));
    pipe.close_in();
    pipe.close_out();
    ASSERT_TRUE(typeid(pipe.auto_close) == typeid(bool));
}

TEST(TestPipeManual, PipeCloseErrorRepr) {
    PipeCloseError e;
    ASSERT_TRUE(typeid(e.repr()) == typeid(std::string));
}

File: RepoTransBench/source_projects/Python/zqqqqz2000_shshsh/tests/test_pipe_manual.py
import pytest
from shshsh.pipe import Pipe, PipeCloseError
import os

def test_pipe_init_and_close(monkeypatch):
    pipe = Pipe()
    # in_fd and out_fd should be open
    assert hasattr(pipe, "in_fd") and hasattr(pipe, "out_fd")
    # Test close_in and close_out
    monkeypatch.setattr(os, "close", lambda x: None)
    pipe.close_in()
    pipe.close_out()
    assert isinstance(pipe.auto_close, bool)

def test_pipe_close_error_repr():
    e = PipeCloseError()
    assert isinstance(repr(e), str)